tests/epics/adpilatus/test_pilatus.py

```diff
@@ -110,7 +97,7 @@ async def _trigger(
 
 
 async def test_hints_from_hdf_writer(test_adpilatus: adpilatus.PilatusDetector):
-    assert test_adpilatus.hints == {"fields": ["test_adpilatus"]}
+    assert test_adpilatus.hints == {"fields": ["test_adpilatus1"]}
```

Comment:
```suggestion
    assert test_adpilatus.hints == {"fields": [test_adpilatus.name]}
```

nit: I think the important check here is that the hinted field for the hdfwriter comes from the device name